### PR TITLE
Windows build warnings

### PR DESCRIPTION
**Pull request by albertony**
Fixed all compiler warnings when building only the command line programs with Windows with Visual Studio 2022 (VC 14.30):
```
"BARRIER_BUILD_GUI": false,
"BARRIER_BUILD_INSTALLER": false
```

I'm quite sure the added numeric type casts are safe with regards to overflow.

The most controversial change is probably the added header `src/lib/common/win32/winapi.h` for just including `Windows.h` - but with preprocessor directives: `WIN32_LEAN_AND_MEAN`, `STRICT` and `NOMINMAX`. Don't know if you think this is a good idea? (And if you have a better name for it...) Main reason is I needed the `NOMINMAX` to avoid conflict with legacy max macro when using `numeric_limits<int>::max()`. There was (and still is outside of the command line "world" I am compiling atm) quite a few instances of the following:

```
#define WIN32_LEAN_AND_MEAN
#include <Windows.h>
```

An alternative could be to define them at CMake level with `add_compile_definitions`, but would then have to be done for every target..

PS: Congrats on your new project. Looking forward to follow its development!

## Contributor Checklist:

* [x] I have created a file in the `doc/newsfragments` directory *IF* it is a
      user-visible change (and make sure to read the `README.md` in that directory)

### CODE REVIEW

File: src/lib/common/win32/winapi.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    This package is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
```
Comment (p12tic) [inline src/lib/common/win32/winapi.h:18]: Let's use macro-based include guards.

File: src/lib/net/SecureUtils.cpp [hunk 2]
```diff
@@ -241,12 +241,11 @@ std::string create_fingerprint_randomart(const std::vector<std::uint8_t>& dgst_r
      * intersects with itself.  Matter of taste.
      */
     const char* augmentation_string = " .o+=*BOX@%&#/^SE";
-    char *p;
```
Comment (p12tic) [inline src/lib/net/SecureUtils.cpp:244]: Separate commit.

File: src/lib/net/SecureUtils.cpp [hunk 3]
```diff
@@ -241,12 +241,11 @@ std::string create_fingerprint_randomart(const std::vector<std::uint8_t>& dgst_r
...
     std::uint8_t field[FLDSIZE_X][FLDSIZE_Y];
     std::size_t i;
     std::uint32_t b;
     int	 x, y;
-    std::size_t len = strlen(augmentation_string) - 1;
+    std::uint8_t len = static_cast<uint8_t>(strlen(augmentation_string) - 1);
```
Comment (p12tic) [inline src/lib/net/SecureUtils.cpp:248]: This one is not obvious. Add a comment above, something like "// avoid compiler warning when comparing len to items in field array"

File: src/lib/platform/MSWindowsScreen.cpp [hunk 4]
```diff
@@ -52,6 +52,7 @@
 #if defined(_MSC_VER)
 #pragma warning(push)
 #pragma warning(disable: 4706) // assignment within conditional
+#pragma warning(disable: 4996) // GetVersionExA was declared deprecated
```
Comment (p12tic) [inline src/lib/platform/MSWindowsScreen.cpp:55]: Separate commit.

**Review by p12tic — CHANGES_REQUESTED** [review]
Thanks a lot for this pull request, warnings are annoying and fixing them is much appreciated.

I agree with everything you've done, it looks great. I'd like the changes to be split into separate commits, like for windows.h header, for the implicit conversion fixes and so on.

**Comment by albertony** [comment]
Sure. Done. Pushed.

**Review by p12tic — APPROVED** [review]
Thanks!

**Comment by p12tic** [comment]
Looks like the `mojave-Release` build has trouble launching lately, maybe some kind of abuse prevention is going on as this project currently does not have much track record of not abusing CI.

I'll merge without CI rerun.

**Pull request merged**